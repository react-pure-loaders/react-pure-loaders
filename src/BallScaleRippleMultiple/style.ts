import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import delay from '../utilities';
import { PRIMARY_COLOR } from '../variables';
import { Grid } from '../types';

const size = '50px';
const ballScaleRippleMultiple = keyframes({
  '0%': {
    opacity: 1,
    transform: 'scale(0.1)',
  },
  '100%': {
    opacity: 0.0,
  },
  '70%': {
    opacity: 0.7,
    transform: 'scale(1)',
  },
});

function ballScaleRipple() {
  const n = 3;
  let i = 0;
  const grid: Grid = {};
  for (; i < 4; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animationDelay: delay('0.2s', n, i - 1),
    };
  }

  return grid;
}

export default (color?: string) => ({
  ...ballScaleRipple(),
  position: 'relative' as 'relative',
  transform: `translateY(-${parseInt(size, 10) / 2})`,
  ' > div': {
    ...globalAnimation(),
    animation: `${ballScaleRippleMultiple} 1.25s 0s infinite cubic-bezier(.21,.53,.56,.8)`,
    border: `2px solid ${color || PRIMARY_COLOR}`,
    borderRadius: '100%',
    height: size,
    left: '-26px',
    position: 'absolute' as 'absolute',
    top: '-2px',
    width: size,

  },
});
