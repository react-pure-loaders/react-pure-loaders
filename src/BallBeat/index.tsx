/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
