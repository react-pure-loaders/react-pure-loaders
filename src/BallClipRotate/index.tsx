/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
