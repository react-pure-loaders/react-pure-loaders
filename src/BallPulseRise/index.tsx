/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
