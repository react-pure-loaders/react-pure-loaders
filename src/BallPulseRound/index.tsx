/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
