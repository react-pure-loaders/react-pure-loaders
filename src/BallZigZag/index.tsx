/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
