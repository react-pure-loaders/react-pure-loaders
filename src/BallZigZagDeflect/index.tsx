/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
