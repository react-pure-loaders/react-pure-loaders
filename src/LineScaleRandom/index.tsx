/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
