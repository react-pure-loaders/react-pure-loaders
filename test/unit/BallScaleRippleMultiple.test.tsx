import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallScaleRippleMultiple from '../../src/BallScaleRippleMultiple';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallScaleRippleMultiple>', () => {
    afterEach(cleanup);

    test('BallScaleRippleMultiple should match snapshot', () => {
        const { container } = render(<BallScaleRippleMultiple loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallScaleRippleMultiple should have default color', () => {
        const { container } = render(<BallScaleRippleMultiple loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('border', `2px solid ${PRIMARY_COLOR}`, { target: '> div' });
    });

    test('BallScaleRippleMultiple should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallScaleRippleMultiple color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('border', `2px solid ${color}`, { target: '> div' });
    });

    test('BallScaleRippleMultiple should have no children', () => {
        const { container } = render(<BallScaleRippleMultiple loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallScaleRippleMultiple should have three children', () => {
        const { container } = render(<BallScaleRippleMultiple loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
