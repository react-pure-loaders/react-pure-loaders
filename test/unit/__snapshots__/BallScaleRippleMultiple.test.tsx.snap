// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallScaleRippleMultiple> BallScaleRippleMultiple should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: scale(0.1);
    -moz-transform: scale(0.1);
    -ms-transform: scale(0.1);
    transform: scale(0.1);
  }

  100% {
    opacity: 0;
  }

  70% {
    opacity: 0.7;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translateY(-25);
  -moz-transform: translateY(-25);
  -ms-transform: translateY(-25);
  transform: translateY(-25);
}

.emotion-0>div:nth-of-type(0) {
  -webkit-animation-delay: -0.80s;
  animation-delay: -0.80s;
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation-delay: -0.60s;
  animation-delay: -0.60s;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation-delay: -0.40s;
  animation-delay: -0.40s;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation-delay: -0.20s;
  animation-delay: -0.20s;
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1.25s 0s infinite cubic-bezier(.21,.53,.56,.8);
  animation: animation-0 1.25s 0s infinite cubic-bezier(.21,.53,.56,.8);
  border: 2px solid #ffffff;
  border-radius: 100%;
  height: 50px;
  left: -26px;
  position: absolute;
  top: -2px;
  width: 50px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
